File: Api/src/Api.Domain/Entities/UserTeamEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities {
    public class UserTeamEntity : BaseEntity
    {
        [Required]
        public int UserId { get; set; }

        [JsonIgnore]
        public UserEntity User { get; set; }

        [Required]
        public int TeamId { get; set; }

        [JsonIgnore]
        public TeamEntity Team { get; set; }
    }
}

File: App/src/app/shared/entities/userTeam.entity.ts
import { TeamEntity } from '../../team/entities/team.entity';
import { UserEntity } from '../../user/entities/user.entity';
import { BaseEntity } from './base.entity';

export interface UserTeamEntity extends BaseEntity {
  userId: number;
  user?: UserEntity;
  teamId: number;
  team?: TeamEntity;
}
